sql/internal/specutil/types.go

```diff
@@ -40,3 +41,42 @@ func PrintType(typ *schemaspec.Type, spec *schemaspec.TypeSpec) (string, error)
 	}
 	return typ.T + mid + suffix, nil
 }
+
+// TypeRegistry is a collection of *schemaspec.TypeSpec.
+type TypeRegistry struct {
+	r []*schemaspec.TypeSpec
+}
+
+// Register adds one or more TypeSpec to the registry.
+func (r *TypeRegistry) Register(specs ...*schemaspec.TypeSpec) error {
+	for _, s := range specs {
+		if _, exists := r.Find(s.T); exists {
+			return fmt.Errorf("specutil: type with T of %q already registered", s.T)
+		}
```

Comment:
If the Name is the Atlas identifier, why checking also the type?

Reply:
I have some concerns on how to handle multiple identifiers that map to the same db type. Namely, how to disambiguate between them. Because I don't have a solution there yet, I'm defensive and enforcing the variant that currently only one identifier  can represent `T`.

Reply:
Why do we care about the database type in the DDL part? Why not allow registering both "int" and "integer" on MySQL, even though both of them map to `int? See https://dev.mysql.com/doc/refman/8.0/en/integer-types.html

> MySQL supports the SQL standard integer types INTEGER (or INT) ...

Reply:
b/c you need to be able to go back and forth from the db to hcl without losing info
if a user used `int` you want it to marshal/unmarshal/inspect the same.

Reply:
If you use `integer` in the DDL (atlas or create stmt), you'll get back an `int` type on inspection.